### PR TITLE
Nix flake

### PR DESCRIPTION
**Pull request by thomascft**
- wip nix flake
- added default app

### CODE REVIEW

**Comment by thomascft** [comment]
I just moved to NixOS and made this flake. Currently it just installs swww to your path.

**Comment by SomeGuyNamedMy** [comment]
might want to add common rust tools like cargo and rust-analyzer as well

**Comment by SomeGuyNamedMy** [comment]
how do you feal about ading a home-manager and nixos module as well?

**Comment by yavko** [comment]
> how do you feal about adding a home-manager and nixos module as well? ive already got the basic stuff for swww set up for the home-manager module.

It won't hurt, its only optional to import

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      };
+
+      overlays = {
+        swww = _: prev: {
+          swww = self.packages.x86_64-linux.swww;
+        };
+        default = self.overlays.swww;
+      };
+      devShells.default = pkgs.mkShell {
+        buildInputs = with pkgs; [ pkg-config libxkbcommon lz4 ];
```
Comment (yavko) [inline flake.nix:35]: add:
`rust`

Comment (yavko) [reply flake.nix:35]: I think the way eww does this is good, so I would take a look 😉 https://github.com/elkowar/eww/blob/master/flake.nix

**Comment by SomeGuyNamedMy** [comment]
im fairly new to git and github, how do you commit to another persons pull request?

**Comment by yavko** [comment]
> im fairly new to git and github, how do you commit to another persons pull request?

You make a pr to their repo on the branch being merged

**Comment by SomeGuyNamedMy** [comment]
thanks, githubs layout doesnt really make that obvious for some reason

**Comment by thomascft** [comment]
I'd be fine with trying to merge the home manager stuff, although I'll probably re factor it into a single file.

**Comment by thomascft** [comment]
> how do you feal about adding a home-manager and nixos module as well? ive already got the basic stuff for swww set up for the home-manager module.

Just added that by hand. I'm compiling and I'll push.

**Comment by Horus645** [comment]
Forgive me, I don't know about this, but looking at other projects, it doesn't seem like having a `flake.nix` file is a standard. Am I wrong about this?

I hate to say this after the work you guys put into this, but the thing is, I have strong reservations against putting something in this repository that I don't really understand or use. It makes everything much more difficult. For example, if a dependency ever changes, I would have to edit this file myself. Ultimately, the reasons are very similar to those of [PR 31](https://github.com/Horus645/swww/pull/31).

**Comment by yavko** [comment]
> Forgive me, I don't know about this, but looking at other projects, it doesn't seem like having a `flake.nix` file is a standard. Am I wrong about this?
> 
> I hate to say this after the work you guys put into this, but the thing is, I have strong reservations against putting something in this repository that I don't really understand or use. It makes everything much more difficult. For example, if a dependency ever changes, I would have to edit this file myself. Ultimately, the reasons are very similar to those of [PR 31](https://github.com/Horus645/swww/pull/31).

No flakes are a standard, a experimental one but very widely used, if you don't want it at the root of the repo you can always put it in a nix directory

**Comment by SomeGuyNamedMy** [comment]
thats fine, this work can also be push towards upstream nixpkgs and home-manager, it just wont be as up to date as with upstream considering how slow home-manager moves

**Comment by yavko** [comment]
> thats fine, this work can also be push towards upstream nixpkgs and home-manager, it just wont be as up to date as with upstream considering how slow home-manager moves

Thats exactly why It would be nice to have as a flake

**Comment by rsHalford** [comment]
I found this merge request over on [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs/pull/202280). So hopefully it gets merged soon!

**Comment by Horus645** [comment]
Just a heads up: be careful that once a new release comes out, your will have to modify the package to install 2 binaries instead of one: `swww` and `swww-daemon`.

**Pull request closed without merging**